syntax_and_semantics/macros/hooks.md

```diff
@@ -1,11 +1,12 @@
 # Hooks
 
-Special macros exist that are invoked in some situations as hooks:
-`inherited`, `included`, `extended` and `method_missing`.
-* `inherited` is invoked at compile-time when a subclass is defined. `@type` is the inheriting type.
-* `included` is invoked at compile-time when a module is included. `@type` is the including type.
-* `extended` is invoked at compile-time when a module is extended. `@type` is the extending type.
-* `method_missing` is invoked at compile-time when a method is not found.
+Special macros exist that are invoked in some situations as hooks, at compile time:
+* `inherited` is invoked when a subclass is defined. `@type` is the inheriting type.
+* `included` is invoked when a module is included. `@type` is the including type.
+* `extended` is invoked when a module is extended. `@type` is the extending type.
+* `method_missing` is invoked when a method is not found.
+* `method_added` is invoked when a new method is defined in the current scope.
+* `finished` is invoked when the current scope is finished compiling.
```

Comment:
"Finished compiling" isn't really right here. It's actually at a very early pass in the compile process.

Reply:
Do you have any suggestions? I guess I kind of think of it as happening after the code is compiled, but of course it can't be.

My only other ideas would be "finished processing"?

Elixir has a similar macro called `__before_compile__` and their documentation describes it as: "@before_compile allow us to inject code into the module when its definition is complete."

How about: `finished` is invoked when the definition of the current scope is complete.

Thoughts?

Reply:
When it runs isn't related to scope at all. I'd probably word is as just "after instance variable types for all classes are known"

Reply:
Updated 👍🏻